sklearn/model_selection/_validation.py

```diff
@@ -1802,6 +1802,23 @@ def _incremental_fit_estimator(
     return np.array(ret).T
 
 
+@validate_params(
+    {
+        "estimator": [HasMethods(["fit", "predict"])],
```

Comment:
Technically, I don't think having ``predict`` is mandatory. You pass pass a callable as ``scoring`` that don't necessarily rely on a ``predict`` method to compute a score. What do think @glemaitre ?